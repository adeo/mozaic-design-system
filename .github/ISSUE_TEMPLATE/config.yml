contact_links:
  - name: Open an issue for Mozaic-React
    url: https://github.com/adeo/mozaic-react/issues
    about: If your request is for the React implementation of Mozaic, please go to the dedicated repository.
  - name: Open an issue for Mozaic-Vue
    url: https://github.com/adeo/mozaic-vue/issues
    about: If your request is for the Vue.js implementation of Mozaic, please go to the dedicated repository.
  - name: Open an issue for Mozaic-WebComponents
    url: https://github.com/adeo/mozaic-web-components/issues
    about: If your request is for the Web Components implementation of Mozaic, please go to the dedicated repository.
