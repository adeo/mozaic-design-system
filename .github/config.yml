helpr:
  opened: 'opened-pull-request'
  merged: 'merged-pull-request'
  rejected: 'rejected-pull-request'
