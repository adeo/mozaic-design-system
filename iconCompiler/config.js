const path = require('path')
const currentDir = process.cwd()

module.exports = {
  iconsOutputPath: {
    svg: path.join(currentDir, 'packages/icons/svg/'),
    react: path.join(currentDir, 'packages/icons/react/'),
    vue: path.join(currentDir, 'packages/icons/vue/'),
  },
}
