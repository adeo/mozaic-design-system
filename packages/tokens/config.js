const CM = require('@mozaic-ds/configuration-manager')
const path = require('path')

const getPath = localRelativePath =>
  path.relative(process.cwd(), `${__dirname}/${localRelativePath}`)

const getBuildPath = path => getPath(path) + '/'

const localSrcPath = CM.getKey('tokens.localTokensSrcPath')

const localTokensExportPath = CM.getKey('tokens.localTokensExportPath')

const source = localSrcPath
  ? [getPath('properties/**/*.json'), `${localSrcPath}properties/**/*.json`]
  : [getPath('properties/**/*.json')]

const setLocalTokensExportPath = dir =>
  localTokensExportPath
    ? `${localTokensExportPath}${dir}/`
    : getBuildPath(`build/${dir}/`)

const config = {
  source,
  platforms: {
    scss: {
      transformGroup: 'scss',
      buildPath: setLocalTokensExportPath('scss'),
      files: [
        {
          destination: '_tokens.scss',
          format: 'scss/map-deep',
          options: {
            showFileHeader: false,
          },
        },
      ],
    },
    android: {
      transformGroup: 'android',
      buildPath: setLocalTokensExportPath('android'),
      files: [
        {
          destination: 'font_dimens.xml',
          format: 'android/fontDimens',
          options: {
            showFileHeader: false,
          },
        },
        {
          destination: 'colors.xml',
          format: 'android/colors',
          options: {
            showFileHeader: false,
          },
        },
      ],
    },
    js: {
      transformGroup: 'js',
      buildPath: setLocalTokensExportPath('js'),
      files: [
        {
          destination: 'tokensObject.js',
          format: 'javascript/module',
          options: {
            showFileHeader: false,
          },
        },
        {
          format: 'javascript/es6',
          destination: 'tokens.js',
          options: {
            showFileHeader: false,
          },
        },
      ],
    },
    ios: {
      transformGroup: 'ios',
      buildPath: setLocalTokensExportPath('ios'),
      files: [
        {
          destination: 'StyleDictionaryColor.h',
          format: 'ios/colors.h',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionaryColor',
          type: 'StyleDictionaryColorName',
          options: {
            showFileHeader: false,
          },
          filter: {
            attributes: {
              category: 'color',
            },
          },
        },
        {
          destination: 'StyleDictionaryColor.m',
          format: 'ios/colors.m',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionaryColor',
          type: 'StyleDictionaryColorName',
          options: {
            showFileHeader: false,
          },
          filter: {
            attributes: {
              category: 'color',
            },
          },
        },
        {
          destination: 'StyleDictionarySize.h',
          format: 'ios/static.h',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionarySize',
          options: {
            showFileHeader: false,
          },
          type: 'float',
          filter: {
            attributes: {
              category: 'size',
            },
          },
        },
        {
          destination: 'StyleDictionarySize.m',
          format: 'ios/static.m',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionarySize',
          type: 'float',
          options: {
            showFileHeader: false,
          },
          filter: {
            attributes: {
              category: 'size',
            },
          },
        },
      ],
    },
  },
}

console.log(JSON.stringify(config, 0, 2))

module.exports = config
