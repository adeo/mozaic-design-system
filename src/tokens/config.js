const path = require('path')

const getPath = localRelativePath =>
  path.relative(process.cwd(), `${__dirname}/${localRelativePath}`)

const getBuildPath = path => getPath(path) + '/'

const config = {
  source: [getPath('properties/**/*.json')],
  platforms: {
    scss: {
      transformGroup: 'scss',
      buildPath: getBuildPath('build/scss/'),
      files: [
        {
          destination: '_tokens.scss',
          format: 'scss/map-deep',
          options: {
            showFileHeader: false,
          },
        },
      ],
    },
    android: {
      transformGroup: 'android',
      buildPath: getBuildPath('build/android/'),
      files: [
        {
          destination: 'font_dimens.xml',
          format: 'android/fontDimens',
          options: {
            showFileHeader: false,
          },
        },
        {
          destination: 'colors.xml',
          format: 'android/colors',
          options: {
            showFileHeader: false,
          },
        },
      ],
    },
    js: {
      transformGroup: 'js',
      buildPath: getBuildPath('build/js/'),
      files: [
        {
          destination: 'tokensObject.js',
          format: 'javascript/module',
          options: {
            showFileHeader: false,
          },
        },
        {
          format: 'javascript/es6',
          destination: 'tokens.js',
          options: {
            showFileHeader: false,
          },
        },
      ],
    },
    ios: {
      transformGroup: 'ios',
      buildPath: getBuildPath('build/ios/'),
      files: [
        {
          destination: 'StyleDictionaryColor.h',
          format: 'ios/colors.h',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionaryColor',
          type: 'StyleDictionaryColorName',
          options: {
            showFileHeader: false,
          },
          filter: {
            attributes: {
              category: 'color',
            },
          },
        },
        {
          destination: 'StyleDictionaryColor.m',
          format: 'ios/colors.m',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionaryColor',
          type: 'StyleDictionaryColorName',
          options: {
            showFileHeader: false,
          },
          filter: {
            attributes: {
              category: 'color',
            },
          },
        },
        {
          destination: 'StyleDictionarySize.h',
          format: 'ios/static.h',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionarySize',
          options: {
            showFileHeader: false,
          },
          type: 'float',
          filter: {
            attributes: {
              category: 'size',
            },
          },
        },
        {
          destination: 'StyleDictionarySize.m',
          format: 'ios/static.m',
          options: {
            showFileHeader: false,
          },
          className: 'StyleDictionarySize',
          type: 'float',
          options: {
            showFileHeader: false,
          },
          filter: {
            attributes: {
              category: 'size',
            },
          },
        },
      ],
    },
  },
}

module.exports = config
