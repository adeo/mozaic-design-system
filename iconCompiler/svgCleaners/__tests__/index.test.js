import config from '../../config'
jest.mock('../../config', () => {
  return {
    sourcePaths: {
      svg: 'iconCompiler/svgCleaners/__tests__/testIcons',
    },
    outputPaths: {
      svg: 'toto',
    },
  }
})
const fs = require('fs')
import svgCleaner from '../index'
const { monochromOptim } = require('../optimizationConfigs')
fs.writeFile = jest.fn((a, b, c, d) => d())

describe('svgCleaner', () => {
  it('return a list of icons', () => {
    config.sourcePaths.svg = 'iconCompiler/svgCleaners/__tests__/testIcons'
    return svgCleaner('svg', monochromOptim).then(res =>
      expect(res).toMatchSnapshot()
    )
  })

  it('return errors if the directory does not exist', () => {
    config.sourcePaths.svg = 'dontexist'
    return svgCleaner('svg', monochromOptim).catch(err =>
      expect(err).toMatchSnapshot()
    )
  })

  it('return errors when the name of an icon is not valid', () => {
    config.sourcePaths.svg = 'iconCompiler/svgCleaners/__tests__/testIconsError'
    return svgCleaner('svg', monochromOptim).catch(err =>
      expect(err).toMatchSnapshot()
    )
  })

  it('return errors when the name of an icon is not valid', () => {
    config.sourcePaths.svg =
      'iconCompiler/svgCleaners/__tests__/testIconsErrorSquareLayer'
    return svgCleaner('svg', monochromOptim).catch(err =>
      expect(err).toMatchSnapshot()
    )
  })
})
