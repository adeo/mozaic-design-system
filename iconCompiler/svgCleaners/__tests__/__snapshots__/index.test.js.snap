// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svgCleaner return a list of icons 1`] = `
Array [
  Object {
    "data": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\"><path d=\\"M8 11.5a1 1 0 01-.71-.29l-5-5a1 1 0 011.42-1.42L8 9.09l4.29-4.3a1 1 0 111.42 1.42l-5 5a1 1 0 01-.71.29z\\"/></svg>",
    "fileName": "Navigation_Arrow_Arrow--Bottom_16px.svg",
    "outputPath": "toto",
  },
  Object {
    "data": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\"><path d=\\"M2 10.5a1 1 0 01.29-.71l5-5a1 1 0 011.42 0l5 5a1 1 0 01-1.42 1.42L8 6.91l-4.29 4.3a1 1 0 01-1.42 0A1 1 0 012 10.5z\\"/></svg>",
    "fileName": "Navigation_Arrow_Arrow--Top_16px.svg",
    "outputPath": "toto",
  },
  Object {
    "data": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\"><path d=\\"M17.59 18.59H6.41a1 1 0 01-1-1V6.41a1 1 0 011-1h11.18a1 1 0 011 1v11.18a1 1 0 01-1 1zm-10.18-2h9.18V7.41H7.41z\\"/><path d=\\"M3 8.39a1 1 0 01-1-1V3a1 1 0 011-1h4.38a1 1 0 010 2H4v3.39a1 1 0 01-1 1zM21 8.38a1 1 0 01-1-1V4h-3.38a1 1 0 110-2H21a1 1 0 011 1v4.38a1 1 0 01-1 1zM7.38 22H3a1 1 0 01-1-1v-4.39a1 1 0 012 0V20h3.38a1 1 0 110 2zM22 21a1 1 0 01-1 1h-4.38a1 1 0 010-2H20v-3.38a1 1 0 112 0z\\"/></svg>",
    "fileName": "Media_FullScreen_24px.svg",
    "outputPath": "toto",
  },
  Object {
    "data": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\"><path d=\\"M12 4a8 8 0 11-8 8 8 8 0 018-8m0-2a10 10 0 1010 10A10 10 0 0012 2z\\"/><path d=\\"M10.72 9.77L14.36 12l-3.64 2.22V9.77M9.73 7a1 1 0 00-1 1v8a1 1 0 001 1 .94.94 0 00.52-.15l6.55-4a1 1 0 000-1.71l-6.55-4A1 1 0 009.73 7z\\"/></svg>",
    "fileName": "Media_Player_24px.svg",
    "outputPath": "toto",
  },
]
`;

exports[`svgCleaner return errors if the directory does not exist 1`] = `"the directory dontexist/**/*.svg is empty, or do not exist"`;

exports[`svgCleaner return errors when the name of an icon is not valid 1`] = `"in Navigation_Arrow_Arrow--Bottom_32px.svg :  found in /16px/ directory. Icon file name is either wrong or the icon is in the wrong directory"`;

exports[`svgCleaner return errors when the name of an icon is not valid 2`] = `"âœ— ERROR : in iconCompiler/svgCleaners/__tests__/testIconsErrorSquareLayer/16px/Navigation_Arrow_Arrow--Top_16px.svg Icon should have a group with a \\"Square\\" id"`;
